Content.Server/_CD/Loadouts/RenameIdSystem.cs

```diff
@@ -42,12 +46,29 @@ private void OnPlayerSpawn(PlayerSpawnCompleteEvent args)
         if (!_inventorySystem.TryGetSlotEntity(player, "id", out var pdaUid))
             return;
 
-        if (TryComp<PdaComponent>(pdaUid, out var pda)
-            && TryComp<RenameIdComponent>(pdaUid, out var rename)
-            && pda.ContainedId is {} id
-            && TryComp<IdCardComponent>(id, out var card))
+        if (!TryComp<PdaComponent>(pdaUid, out var pda) ||
+            !TryComp<RenameIdComponent>(pdaUid, out var rename) ||
+            pda.ContainedId is not { } id ||
+            !TryComp<IdCardComponent>(id, out var card))
+            return;
+
+        UpdateIdByComp((id, card), rename);
+    }
+
+    private void UpdateIdByComp(Entity<IdCardComponent> id, RenameIdComponent comp)
+    {
+        if (comp.NewIcon is { } icon)
         {
-            _idCardSystem.TryChangeJobTitle(id, Loc.GetString(rename.Value), card);
+            id.Comp.JobIcon = icon; // TryChangeJobTitle dirties the ID for us
+            if (!TryComp<StationRecordKeyStorageComponent>(id, out var keyStorage) ||
+                keyStorage.Key is not { } key ||
+                !_records.TryGetRecord<GeneralStationRecord>(key, out var record))
+                return;
+
+            record.JobIcon = icon;
+
+            _records.Synchronize(key);
```

Comment:
```suggestion
            if (TryComp<StationRecordKeyStorageComponent>(id, out var keyStorage) &&
                keyStorage.Key is { } key &&
                _records.TryGetRecord<GeneralStationRecord>(key, out var record))
            {
                record.JobIcon = icon;
                _records.Synchronize(key);
            }
```
This way the rest of the function will run even if the record doesn't exist for some reason.
Without this, there's a conceivable codepath where the ID isn't dirtied and things go wrong.